otherlibs/cram/bin/run.ml

```diff
@@ -2,6 +2,40 @@ open Stdune
 open Cmdliner
 open Import.Let_syntax
 
+module Temp_dir : sig
+  type t
+
+  val create : for_script:string -> t
+
+  val open_file : t -> suffix:string -> string * out_channel
+
+  val file : t -> suffix:string -> string
+end = struct
+  type t = string
+
+  let prng = lazy (Random.State.make_self_init ())
+
+  let temp_file_name temp_dir prefix suffix =
+    let rnd = Random.State.bits (Lazy.force prng) land 0xFFFFFF in
+    Filename.concat temp_dir (Printf.sprintf "%s%06x%s" prefix rnd suffix)
+
+  let create ~for_script =
+    let dir = Filename.get_temp_dir_name () in
+    let t = temp_file_name dir ".dune.cram." (Filename.basename for_script) in
+    let path = Path.External.of_string t in
+    Path.External.mkdir_p path;
+    at_exit (fun () -> Path.rm_rf ~allow_external:true (Path.external_ path));
```

Comment:
@jeremiedimino I've added this clean up because the old code also had this clean up, but I don't see why it's necessary. I recall that in other instances we never bothered to clear temp dirs.

Reply:
Hmm, if we don't cleanup temp dirs, what will clean them up? We definitely don't want to leave them around after the application has exited.

BTW, we recently added `Temp` module in `Stdune`, could you put this code there?

Reply:
> Hmm, if we don't cleanup temp dirs, what will clean them up? 

I assumed the OS would. 

> BTW, we recently added Temp module in Stdune, could you put this code there?

Will do